util/topgen/templates/plic_all_irqs_test.c.tpl

```diff
@@ -197,10 +198,9 @@ static void peripheral_irqs_trigger(void) {
   ${indent}  LOG_INFO("Triggering ${p.inst_name} IRQ %d.", irq);
   ${indent}  CHECK_DIF_OK(dif_${p.name}_irq_force(&${p.inst_name}, irq, true));
 
-  ${indent}  // TODO: Make race-condition free
-  ${indent}  CHECK(${p.name}_irq_serviced == irq,
-  ${indent}        "Incorrect ${p.inst_name} IRQ serviced: exp = %d, obs = %d", irq,
-  ${indent}        ${p.name}_irq_serviced);
+  ${indent}  // This avoids a race where *irq_serviced is read before
+  ${indent}  // entering the CSR.
+  ${indent}  IBEX_SPIN_FOR(${p.name}_irq_serviced == irq, 1);
```

Comment:
do you by chance mean ISR instead of CSR? maybe I misunderstood what CSR means in this context ;)...

Reply:
I will make another PR to fix that. Good catch MIchael.